### PR TITLE
Update to poetry and python 3.10

### LINKED ISSUE #21: Use Gunicorn HTTP server
Currently using Flask dev server, which is strongly recommended against.

### LINKED ISSUE #22: update to poetry
Update the build to use poetry and maybe python 3.10.

### PR DESCRIPTION
**Pull request by corviday**
Updates to `poetry`, python 3.10, and Ubuntu 22.04.

As part of the update, `connexion` had to be updated to 3.x because `connexion` 2.x uses a version of `Flask` incompatible with python 3.10. `Connexion` 3.x requires an asgi framework instead of the wsgi one we'd been using, but this change turned out to be fairly straightforward. `Connexion` had previously been providing a wsgi-style response loop; upgrading to the new `connexion` got us an asgi-style response loop pretty much for free. All we really had to do was switch to using `uvicorn` (which `connexion` offers integration with as an extra) as the server.

[Demo](https://beehive.pacificclimate.org/wx-files/app/).

resolves #22 
resolves #21

### CODE REVIEW

**Comment by corviday** [comment]
connexion [installs starlette](https://github.com/spec-first/connexion/blob/a930303faa9e10603cdd820c497303b2e52d8253/pyproject.toml#L56), so I am going to proceed on the assumption that we don't need a separate asgi framework.

File: .github/workflows/python-ci.yml [hunk 1]
```diff
@@ -5,27 +5,24 @@ on: push
 jobs:
   test:
 
-    runs-on: ubuntu-20.04
-    strategy:
-      matrix:
-        python-version: [3.6, 3.7, 3.8]
+    runs-on: ubuntu-22.04
```
Comment (rod-glover) [inline .github/workflows/python-ci.yml:8]: Only one Python ver? It is possible (but more work :(( ) to use more than one python interpreter in Poetry. Maybe put that in a follow-on issue?

File: docker/Dockerfile [hunk 2]
```diff
@@ -14,20 +14,21 @@ RUN apt-get update && \
...
 EXPOSE 8000
 
-ENV FLASK_APP wxfs.wsgi
+ENV FLASK_APP wxfs.asgi
 ENV FLASK_ENV development
 ENV LC_ALL C.UTF-8
 ENV LANG C.UTF-8
 
-CMD flask run -p 8000 -h 0.0.0.0 --no-reload
+CMD poetry run uvicorn --port 8000 --host 0.0.0.0 --reload wxfs.asgi:connexion_app
```
Comment (rod-glover) [inline docker/Dockerfile:34]: Yay, sooo much better to serve it with something proper. I can't believe I didn't use Gunicorn before.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = "^3.10"
+python-dateutil = "^2.9.0.post0"
+psycopg2 = "^2.9.9"
+Flask = "^3.0.3"
+Flask-Cors = "^5.0.0"
+Flask-SQLAlchemy = "^3.1.1"
+connexion = {extras = ["swagger-ui", "flask", "uvicon"], version = "^3.1.0"}
```
Comment (rod-glover) [inline pyproject.toml:16]: "`uvicon`" or "`uvicorn`"??

File: setup.py
Comment (rod-glover) [inline setup.py:1]: Everything in `setup.py` can (and should) be transferred into `pyproject.toml`, then `setup.py` consigned to the dustbin of history. Example in Met Data Portal: https://github.com/pacificclimate/station-data-portal-backend/blob/master/pyproject.toml

**Review by rod-glover — CHANGES_REQUESTED** [review]
Nice clean job.

`setup.py` needs to die -- see comment below.

It seemed to take a very long time to load on the initial load. Spinner going. Second time was tolerable, about 3 s.  This might bear looking into?

**Comment by corviday** [comment]
I have gotten rid of `setup.py`. 

I think I saw a very slow page load at some point, but I can't reproduce it now.

**Review by rod-glover — APPROVED** [review]
Great, nice work!

**Pull request merged**